components/eam/src/physics/cam/shoc_intr.F90

```diff
@@ -348,46 +356,73 @@ subroutine shoc_init_e3sm(pbuf2d, dp1_in)
     prer_evap_idx   = pbuf_get_index('PRER_EVAP')
     qrl_idx         = pbuf_get_index('QRL')
     cmfmc_sh_idx    = pbuf_get_index('CMFMC_SH')
-    tke_idx         = pbuf_get_index('tke')   
+    tke_idx         = pbuf_get_index('tke')
     vmag_gust_idx = pbuf_get_index('vmag_gust')
- 
+
     if (is_first_step()) then
-      call pbuf_set_field(pbuf2d, wthv_idx, 0.0_r8) 
-      call pbuf_set_field(pbuf2d, tkh_idx, 0.0_r8) 
-      call pbuf_set_field(pbuf2d, tk_idx, 0.0_r8) 
+      call pbuf_set_field(pbuf2d, wthv_idx, 0.0_r8)
+      call pbuf_set_field(pbuf2d, tkh_idx, 0.0_r8)
+      call pbuf_set_field(pbuf2d, tk_idx, 0.0_r8)
       call pbuf_set_field(pbuf2d, fice_idx, 0.0_r8)
       call pbuf_set_field(pbuf2d, tke_idx, tke_tol)
       call pbuf_set_field(pbuf2d, alst_idx, 0.0_r8)
       call pbuf_set_field(pbuf2d, aist_idx, 0.0_r8)
-      
+
       call pbuf_set_field(pbuf2d, vmag_gust_idx,    1.0_r8)
-      
+
     endif
-    
+
     if (prog_modal_aero) then
        ! Turn off modal aerosols and decrement edsclr_dim accordingly
        call rad_cnst_get_info(0, nmodes=nmodes)
- 
+
        do m = 1, nmodes
           call rad_cnst_get_mode_num_idx(m, lptr)
           lq(lptr)=.false.
           edsclr_dim = edsclr_dim-1
- 
+
           call rad_cnst_get_info(0, m, nspec=nspec)
           do l = 1, nspec
              call rad_cnst_get_mam_mmr_idx(m, l, lptr)
              lq(lptr)=.false.
              edsclr_dim = edsclr_dim-1
           end do
        end do
- 
+
        !  In addition, if running with MAM, droplet number is transported
        !  in dropmixnuc, therefore we do NOT want SHOC to apply transport
        !  tendencies to avoid double counted.  Else, we apply tendencies.
        call cnst_get_ind('NUMLIQ',ixnumliq)
        lq(ixnumliq) = .false.
        edsclr_dim = edsclr_dim-1
-    endif 
+    endif
+
+    !SHOC needs all its water constituents in terms of "dry" mmr
+    !but the state vector has all its scalars in terms of "wet" mmr
+    !identify scalar which need conversion between wet and dry mmrs
+    do icnst = 1, pcnst
+       lq_dry_wet_cnvr(icnst) = .false. ! lets assume .false. by default
```

Comment:
didn't you already initialize the whole array to false? This seems redundant?

Reply:
You are right, this is redundant. I can remove it but I kept it there for safety to guard against any change in this variable between its initialization to this code block.

Reply:
@PeterCaldwell ,are you okay with this PR now?  It finally passed the AT so I think we can get it merged, but before I start that process I wanted to make sure you are happy with the response...